src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,154 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React from 'react'
+import {DataLink} from '../../../../components/ui/DbLink'
+import {Event, Events} from '../../../../event'
+import {Analyser} from '../../../core/Analyser'
+import {filter} from '../../../core/filter'
+import {dependency} from '../../../core/Injectable'
+import BrokenLog from '../../../core/modules/BrokenLog'
+import Checklist, {Requirement, Rule} from '../../../core/modules/Checklist'
+import {Data} from '../../../core/modules/Data'
+import {DISPLAY_ORDER} from './DISPLAY_ORDER'
+
+interface ImmortalSacrificeWindow {
+	startTime: number,
+	endTime?: number,
+	/**
+	 * If this window was closed, by consuming the stacks, or it expired
+	 */
+	closed: boolean,
+	stackCount: number,
+	consumed: boolean,
+}
+
+export default class ImmortalSacrifice extends Analyser {
+
+	static override handle = 'immortalsacrifice'
+	static override title = t('rpr.immortal-sacrifice.title')`Immortal Sacrifice stacks consumed`
+	static override debug = false
+
+	@dependency private brokenLog!: BrokenLog
+	@dependency private data!: Data
+	@dependency private checklist!: Checklist
+
+	private windows: ImmortalSacrificeWindow[] = []
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeStack)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeExpiration)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.PLENTIFUL_HARVEST.id), this.onPlentifulHarvest)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onImmortalSacrificeStack(event: Events['statusApply']) {
+		let window = this.getOpenWindow()
+
+		if (window === null) {
```

Comment:
This will never be true.
```suggestion
		if (window == null) {
```